clients/src/main/java/org/apache/kafka/common/requests/AbstractRequest.java

```diff
@@ -27,31 +27,37 @@
 
     public static abstract class Builder<T extends AbstractRequest> {
         private final ApiKeys apiKey;
-        private final Short desiredVersion;
+        private final short minVersion;
+        private final short maxVersion;
```

Comment:
The previous approach was to differentiate between "no version requested" and "desired version requested". We lose that here, but perhaps it doesn't matter?

Reply:
We still have it, it's just that "no version requested" is treated as "all versions are allowed."